src/CovidAppointmentTable.js

```diff
@@ -79,20 +88,34 @@ export default function CovidAppointmentTable() {
 		onlyShowAvailable
 	);
 
-	return (
-		<Loader loaded={!!data && data.length > 0}>
+  return (
+    <>
+      <div id="progress" role="progressbar" aria-valuetext={ready ? 'loaded' : 'waiting'}>
+	<Loader loaded={ready} />
+      </div>
+
+      {errorMessage && (<div role="alert">{errorMessage}</div>)}
+
+      <section aria-live="polite" aria-describedby="progress" aria-busy={!ready}>
 			<FormControlLabel
 				control={
-					<Switch
+			    <Switch
+                              role="switch"
 						checked={onlyShowAvailable}
 						onChange={(event) => setOnlyShowAvailable(event.target.checked)}
 					/>
 				}
 				label="Only show locations with available appointments"
 			/>
+        {ready && formattedData.length === 0 && (
+          <div role="status">
+            <p>No appointments found.</p>
+          </div>
+        )}
+            <div role="list">
 			{formattedData.map((entry) => {
 				return (
-					<div className={classes.cardBox}>
+				  	<div role="listitem" key={`${entry.location}-${entry.city}`} className={classes.cardBox}>
```

Comment:
bad news - we have at least one instance (2 Price Choppers in Worcester) that make this key not unique :(

Reply:
interesting, ok!  I'll see if I can find (or derive) a more unique key for each list item.